testing/ostest/mutex.c

```diff
@@ -103,11 +116,172 @@ static void *thread_func(FAR void *parameter)
   return NULL; /* Non-reachable -- needed for some compilers */
 }
 
+static void *moved_mutex_thread_func(FAR void *parameter)
+{
+  FAR mutex_thread_args_t *args = (FAR mutex_thread_args_t *)parameter;
+
+  int id = args->thread_id;
+  int ndx = id - 1;
+  int i;
+  unsigned long local_loops = 0;
+
+  for (local_loops = 0; local_loops < NLOOPS; local_loops++)
+    {
+      int status = pthread_mutex_lock(args->mutex_ptr);
+      if (status != 0)
+        {
+          printf("ERROR moved mutex thread %d: pthread_mutex_lock failed, "
+                 "status=%d\n", id, status);
+          ASSERT(false);
+        }
+
+      if (*(args->mutex_value) == 1)
+        {
+          printf("ERROR moved mutex thread=%d: "
+                 "moved_mutex_value should be zero, "
+                 "instead moved_mutex_value=%d\n",
+                 id, *(args->mutex_value));
+          ASSERT(false);
+          args->nerrors[ndx]++;
+        }
+
+      *(args->mutex_value) = 1;
+      for (i = 0; i < 10; i++)
```

Comment:
Above, in the outer loop, we are using NLOOPS define; should there be a similar define for this loop, e.g., `NYIELDS`?